### PR TITLE
Update Sync Status Message for Application Volumes (PVCs)

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
This patch addresses the issue where the "last synced status" for Application volumes (PVCs) shows "No data available" in the UI for Discovered apps. Reworded the message to "Not Applicable for Sync Deployments" for better clarity.

https://bugzilla.redhat.com/show_bug.cgi?id=2294374

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
Currently i am testing this patch

**Comment by SanjalKatiyar** [comment]
was it approved by UX/Doc team ??

**Comment by Chandan141996** [comment]
How is it that an app volume sync status is available but not a timestamp? Is this metric not available currently? If we can only report sync details for kube objects, I suggest removing the application sync row entirely.

@TimothyAsirJeyasing @GowthamShanmugam

**Comment by TimothyAsirJeyasing** [comment]
> How is it that an app volume sync status is available but not a timestamp? Is this metric not available currently? If we can only report sync details for kube objects, I suggest removing the application sync row entirely.
> 
> @TimothyAsirJeyasing @GowthamShanmugam

Sure as per our discussion i am removing the row entirely for MDR

**Comment by TimothyAsirJeyasing** [comment]
Screenshot from 2024-07-23 21-40-57

File: packages/mco/components/protected-applications/components.tsx [hunk 1]
```diff
@@ -362,36 +372,60 @@ export const EventsDetails: React.FC<ExpandableComponentProps> = ({
...
+  application,
   syncStatusInfo,
 }) => {
   const { t } = useCustomTranslation();
-
-  const syncType = [t('Application volumes (PVCs)'), t('Kubernetes objects')];
-  const { icon: volIcon, title: volTitle } = replicationHealthMap(
-    syncStatusInfo.volumeReplicationStatus,
-    t
+  const [drPolicies, drPoliciesLoaded] = useK8sWatchResource<DRPolicyKind[]>(
```
Comment (GowthamShanmugam) [inline packages/mco/components/protected-applications/components.tsx:379]: No need to read the drpolicy info here, list-page.tsx is already reading the drpolicy info. All we need to do is introduce a replicationType prop in the ExpandableComponentProps type, and pass the replication type from list-page.tsx to components.tsx

File: packages/mco/components/protected-applications/components.tsx [hunk 2]
```diff
@@ -363,35 +368,47 @@ export const EventsDetails: React.FC<ExpandableComponentProps> = ({
...
+  const syncType = [];
+  const syncStatus = [];
+  const lastSyncOn = [];
 
-  const syncType = [t('Application volumes (PVCs)'), t('Kubernetes objects')];
-  const { icon: volIcon, title: volTitle } = replicationHealthMap(
-    syncStatusInfo.volumeReplicationStatus,
-    t
-  );
+  if (replicationType === REPLICATION_TYPE.SYNC) {
```
Comment (TimothyAsirJeyasing) [inline packages/mco/components/protected-applications/components.tsx:378]: it should be replicationType !== REPLICATION_TYPE.SYNC
to skip for MDR not for RDR right ?

Comment (GowthamShanmugam) [reply packages/mco/components/protected-applications/components.tsx:378]: yes use ASYNC

File: packages/mco/components/protected-applications/list-page.tsx [hunk 3]
```diff
@@ -118,6 +118,15 @@ const ProtectedAppsTableRow: React.FC<
...
     ExpandableComponentsMap[expandableComponentType];
 
+  const volumesSchedulingInterval = Array.isArray(extraProps.drPolicies)
+    ? extraProps.drPolicies.find((policy) => getName(policy) === appName)?.spec
+        ?.schedulingInterval
+    : undefined;
+
+  const replicationType = volumesSchedulingInterval
+    ? getReplicationType(volumesSchedulingInterval)
+    : undefined;
```
Comment (GowthamShanmugam) [inline packages/mco/components/protected-applications/list-page.tsx:128]:
```suggestion

```

Comment (GowthamShanmugam) [reply packages/mco/components/protected-applications/list-page.tsx:128]: let's add replicationType part of SyncStatusInfo and pass it from updateSyncStatus function

File: packages/mco/components/protected-applications/list-page.tsx [hunk 4]
```diff
@@ -223,6 +232,7 @@ const ProtectedAppsTableRow: React.FC<
             <ExpandableComponent
               application={application}
               syncStatusInfo={syncStatusInfo}
+              replicationType={replicationType}
```
Comment (GowthamShanmugam) [inline packages/mco/components/protected-applications/list-page.tsx:235]:
```suggestion
```

File: packages/mco/components/protected-applications/list-page.tsx [hunk 5]
```diff
@@ -307,7 +317,7 @@ export const ProtectedApplicationsListPage: React.FC = () => {
       composableTableProps={{
         columns: getHeaderColumns(t),
         RowComponent: ProtectedAppsTableRow,
-        extraProps: { launcher, navigate, syncStatus },
+        extraProps: { launcher, navigate, syncStatus, drPolicies },
```
Comment (GowthamShanmugam) [inline packages/mco/components/protected-applications/list-page.tsx:320]:
```suggestion
        extraProps: { launcher, navigate, syncStatus },
```

File: packages/mco/components/protected-applications/utils.tsx [hunk 6]
```diff
@@ -145,6 +145,7 @@ export type SyncStatusInfo = {
   volumeLastGroupSyncTime: string;
   kubeObjectReplicationStatus: VOLUME_REPLICATION_HEALTH;
   kubeObjectLastProtectionTime: string;
+  replicationType?: REPLICATION_TYPE;
```
Comment (GowthamShanmugam) [inline packages/mco/components/protected-applications/utils.tsx:148]:
```suggestion
  replicationType: REPLICATION_TYPE;
```

**Review by GowthamShanmugam — APPROVED** [review]

**Pull request merged**